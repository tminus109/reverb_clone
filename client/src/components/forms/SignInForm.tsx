import React, { useEffect, useState } from "react";
import { useTokenContext } from "../../context/TokenContext";
import handleSignIn from "../../services/handleSignIn";

// eslint-disable-next-line react/function-component-definition
function SignInForm() {
  const [email, setEmail] = useState<string>("");
  const [password, setPassword] = useState<string>("");
  const [invalidEmail, setInvalidEmail] = useState<string>("");
  const [invalidPassword, setInvalidPassword] = useState<string>("");
  const [message, setMessage] = useState<string>("");
  const { setToken } = useTokenContext();
  const controller = new AbortController();

  useEffect(() => () => controller.abort());

  return (
    <div className="signin_form">
      <h1>Sign in</h1>
      <form
        onSubmit={(e) => {
          handleSignIn(
            e,
            email,
            password,
            setInvalidEmail,
            setInvalidPassword,
            setMessage,
            setToken,
            controller.signal
          );
        }}
      >
        <label htmlFor="email">
          Email:
          <input
            id="email"
            data-testid="email"
            type="text"
            required
            value={email}
            onChange={(e: React.FormEvent<HTMLInputElement>) => {
              setEmail((e.target as HTMLInputElement).value);
            }}
          />
          {invalidEmail && <p>{invalidEmail}</p>}
        </label>
        <label htmlFor="password">
          Password:
          <input
            id="password"
            data-testid="password"
            type="text"
            value={password}
            onChange={(e: React.FormEvent<HTMLInputElement>) => {
              setPassword((e.target as HTMLInputElement).value);
            }}
          />
          {invalidPassword && <p>{invalidPassword}</p>}
        </label>
        <input type="submit" />
        <div className="message">{message && <p>{message}</p>}</div>
      </form>
    </div>
  );
}

export default SignInForm;
