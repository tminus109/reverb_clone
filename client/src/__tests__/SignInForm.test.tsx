import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SignInForm from "../components/forms/SignInForm";

test("Sign-in form renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SignInForm />, div);
});

test("Snapshot test", () => {
  const tree = renderer.create(<SignInForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
