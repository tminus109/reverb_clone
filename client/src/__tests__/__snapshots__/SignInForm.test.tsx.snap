// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div
  className="signin_form"
>
  <h1>
    Sign in
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="email"
    >
      Email:
      <input
        id="email"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      
    </label>
    <label
      htmlFor="password"
    >
      Password:
      <input
        id="password"
        onChange={[Function]}
        type="text"
        value=""
      />
      
    </label>
    <input
      type="submit"
    />
    <div
      className="message"
    >
      
    </div>
  </form>
</div>
`;
